goal_src/jak2/engine/common_objs/projectile.gc

```diff
@@ -7,3 +7,720 @@
+(defmethod spawn-impact-particles projectile-bounce ((obj projectile-bounce))
+  "Spawns associated particles with the projectile if applicable"
+  (ja-post)
+  0
+  (none)
+  )
+
+(defun projectile-bounce-move ((arg0 projectile-bounce))
+  (seek-toward-heading-vec! (-> arg0 root-override) (-> arg0 root-override transv) 131072.0 (seconds 0.1))
+  (quaternion*! (-> arg0 root-override quat) (-> arg0 root-override quat) (-> arg0 tumble-quat))
+  (projectile-move-fill-all-dirs arg0)
+  (none)
+  )
+
+(defun projectile-bounce-reaction ((arg0 control-info) (arg1 collide-query) (arg2 vector) (arg3 vector))
+  (cshape-reaction-update-state arg0 arg1 arg3)
+  (let ((f30-0 (vector-dot arg3 (-> arg0 surface-normal)))
+        (s2-0 (new 'stack-no-clear 'vector))
+        )
+    (vector-float*! s2-0 (-> arg0 surface-normal) (* f30-0 (rand-vu-float-range 1.6 2.2)))
+    (vector-! arg2 arg3 s2-0)
+    )
+  (-> arg0 status)
+  )
+
+(defmethod play-impact-sound! projectile-bounce ((obj projectile-bounce))
+  "Plays impact sound
+       :virtual"
+  (let* ((a2-0 (-> obj root-override))
+         (v1-0 (-> a2-0 status))
+         )
+    (if (logtest? v1-0 (cshape-moving-flags t-surface))
+        (vector-float*! (-> a2-0 transv) (-> a2-0 transv) 0.6)
+        )
+    (when (and (logtest? v1-0 (cshape-moving-flags csmf11))
+               (>= (- (-> self clock frame-counter) (-> obj played-bounce-time)) (seconds 0.3))
+               )
+      (set! (-> obj played-bounce-time) (-> self clock frame-counter))
+      (sound-play "dark-shot-bounc")
+      )
+    )
+  )
+
+;; WARN: Return type mismatch int vs none.
+(defmethod init-collision! projectile-bounce ((obj projectile-bounce))
+  "Init the [[projectile]]'s [[collide-shape]]"
+  (let ((s5-0 (new 'process 'collide-shape-moving obj (collide-list-enum hit-by-player))))
+    (set! (-> s5-0 dynam) (copy *standard-dynamics* 'process))
+    (set! (-> s5-0 reaction)
+          (the-as (function control-info collide-query vector cshape-moving-flags) projectile-bounce-reaction)
+          )
+    (set! (-> s5-0 no-reaction) nothing)
+    (set! (-> s5-0 penetrate-using) (penetrate jak-dark-shot))
+    (let ((v1-7 (new 'process 'collide-shape-prim-sphere s5-0 (the-as uint 0))))
+      (set! (-> v1-7 prim-core action) (collide-action solid))
+      (set-vector! (-> v1-7 local-sphere) 0.0 0.0 0.0 819.2)
+      (set! (-> s5-0 total-prims) (the-as uint 1))
+      (set! (-> s5-0 root-prim) v1-7)
+      )
+    (set! (-> s5-0 nav-radius) (* 0.75 (-> s5-0 root-prim local-sphere w)))
+    (let ((v1-10 (-> s5-0 root-prim)))
+      (set! (-> s5-0 backup-collide-as) (-> v1-10 prim-core collide-as))
+      (set! (-> s5-0 backup-collde-with) (-> v1-10 prim-core collide-with))
+      )
+    (set! (-> s5-0 max-iteration-count) (the-as uint 2))
+    (set! (-> s5-0 event-self) 'touched)
+    (set! (-> obj root-override) s5-0)
+    )
+  (let ((t9-3 (method-of-object (-> obj root-override) collide-shape-method-47)))
```

Comment:
Looks like this method is missing types. I think the type of the argument is `collide-spec`.
A good name for the method might be `set-collide-with`